mentor_whirlpool/students_handles.py

```diff
@@ -0,0 +1,156 @@
+    async with bot.retrieve_data(message.from_user.id, message.chat.id) as data:
+        data['subject'] = message.text
+
+    await bot.set_state(message.from_user.id, StudentStates.add_work_flag, message.chat.id)
+    await bot.send_message(message.chat.id, "Введите название работы:")
+
+
+@bot.message_handler(state=StudentStates.add_work_flag)
+async def save_request(message):
+    entered_topic = message.text
+    student_dict = dict()
+
+    async with bot.retrieve_data(message.from_user.id, message.chat.id) as data:
+        student_dict = {'name': message.from_user.username, 'chat_id': message.chat.id, 'subjects': [data['subject']],
+                        'description': entered_topic}
+
+    db = Database()
+    await db.add_course_work(student_dict)
+
+    await bot.delete_state(message.from_user.id, message.chat.id)
+    await bot.send_message(message.chat.id, "Работа успешно добавлена! Ожидайте ответа ментора.")
+
+
+@bot.message_handler(func=lambda msg: msg.text == 'Мои запросы')
+async def my_requests(message):
+    db = Database()
+    id = await db.get_students(chat_id=message.chat.id)
+    print(message.chat.id)
+    if not id:
+        await bot.send_message(message.chat.id, f"Пока у вас нет запросов. Скорее добавьте первый!")
+        await add_request(message)
+    else:
+        print(id)
+        student_request = await db.get_course_works(student=id[0]['id'])
+        if not student_request:
+            await bot.send_message(message.chat.id, f"Пока у вас нет запросов.")
+            await add_request(message)
+        else:
+            for course_work in student_request:
+                await bot.send_message(message.chat.id,
+                                       f"*Работа №{course_work['id']}*\nПредмет: {course_work['subjects'][0]}\n"
+                                       f"Тема работы: {course_work['description']}",
+                                       parse_mode='Markdown')
+
+
+@bot.message_handler(func=lambda msg: msg.text == 'Удалить запрос')
+async def remove_request(message):
+    db = Database()
+    id = await db.get_students(chat_id=message.chat.id)
+
+    student_request = await db.get_course_works(student=id[0]['id'])
+
+    markup = types.InlineKeyboardMarkup(row_width=1)
+    for course_work in student_request:
+        markup.add(
+            types.InlineKeyboardButton(course_work['description'],
+                                       callback_data=f"delete_request_{course_work['id']}"))
+
+    await bot.send_message(message.chat.id, "Выберите работу, которую хотите удалить из списка запросов: ",
+                           reply_markup=markup)
+
+
+@bot.message_handler(func=lambda msg: msg.text == 'Хочу стать ментором')
+async def mentor_resume(message):
+    db = Database()
+    admins = await db.get_admins()
+
+    for admin in admins:
+        admin_chat_id = admin['chat_id']
+        await bot.send_message(admin_chat_id, f"Пользователь @{message.from_user.username} хочет стать ментором.")
```

Comment:
Я не уверен что это хорошая идея отсылать заявку всем админам, тогда студенту будет много стука в личку. Наверное лучше взять рандомного
```suggestion
    admin_chat_id = random.choice(admins)['chat_id']
    await bot.send_message(admin_chat_id, f"Пользователь @{message.from_user.username} хочет стать ментором.")
```
Но если всё-таки отсылать всем, то лучше так:
```suggestion
    await gather(*[bot.send_message(adm['chat_id'], f"Пользователь @{message.from_user.username} хочет стать ментором.") for adm in await db.get_admins()])
```
Второй вариант может быть чуть менее читаем, зато значительно быстрее, потому что запросы отправляются асинхронно всем, а не с ожиданием успешной отправки каждому